src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -347,96 +359,135 @@ public void Dispose()
 
-                var reposToReturn = new List<IRepository>();
+        // Establish a connection between organizations and their projects.
+        Parallel.ForEach(organizations, organization =>
+        {
+            var projects = _azureHierarchy.GetProjectsAsync(organization).Result;
 
-                // organizationAndProjects include all relevant information to get repos.
-                Parallel.ForEach(organizationsAndProjects, organizationAndProject =>
-                {
-                    try
-                    {
-                        // Making a connection can throw.
-                        var connection = AzureClientProvider.GetConnectionForLoggedInDeveloper(organizationAndProject.Key.AccountUri, azureDeveloperId);
-                        var criteria = new GitPullRequestSearchCriteria();
-                        criteria.CreatorId = connection.AuthorizedIdentity.Id;
+            if (projects.Count != 0)
+            {
+                organizationsAndProjects.Add(organization, projects);
+            }
+        });
 
-                        Parallel.ForEach(organizationAndProject.Value, project =>
-                        {
-                            reposToReturn.AddRange(GetRepos(connection, project, criteria, shouldIgnorePRDateOld));
-                        });
-                    }
-                    catch (Exception e)
-                    {
-                        Providers.Log.Logger()?.ReportError("DevHomeRepository", e);
-                    }
+        var reposToReturn = new List<IRepository>();
+
+        // organizationAndProjects include all relevant information to get repos.
+        Parallel.ForEach(organizationsAndProjects, organizationAndProject =>
+        {
+            try
+            {
+                // Making a connection can throw.
+                var connection = AzureClientProvider.GetConnectionForLoggedInDeveloper(organizationAndProject.Key.AccountUri, azureDeveloperId);
+                var criteria = new GitPullRequestSearchCriteria();
+                criteria.CreatorId = connection.AuthorizedIdentity.Id;
+
+                Parallel.ForEach(organizationAndProject.Value, project =>
+                {
+                    reposToReturn.AddRange(GetRepos(connection, project, criteria, shouldIgnorePRDate));
                 });
+            }
+            catch (Exception e)
+            {
+                Providers.Log.Logger()?.ReportError("DevHomeRepository", e);
+            }
+        });
 
-                return new RepositoriesResult(reposToReturn);
+        // Figure out what server the organizations are from. if the user did not supply one.
+        if (string.IsNullOrEmpty(_serverToSearch))
+        {
+            HashSet<string> servers = new();
+            foreach (var organization in organizations)
+            {
+                if (organization.AccountUri.OriginalString.Contains("dev.azure.com"))

```

Comment:
Is this supposed to be _defaultSearchServerName?

Reply:
Yes.  Thank you for the catch.